zcash_client_sqlite/src/wallet/init/migrations/full_account_ids.rs

```diff
@@ -69,6 +69,7 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
                     (account_type = {account_type_imported} AND hd_seed_fingerprint IS NULL AND hd_account_index IS NULL)
                 )
             );
+            CREATE UNIQUE INDEX hd_account ON accounts_new (hd_seed_fingerprint, hd_account_index);
```

Comment:
I don't understand the full subtleties of `UNIQUE INDEX` vs `CONSTRAIN UNIQUE` and why you'd choose one over the other. AFAICT:
- They both produce an index.
- `UNIQUE INDEX` can be removed and added without altering the table (and I guess `CREATE UNIQUE INDEX` fails if the table contains non-unique data?)
- `UNIQUE INDEX` can include expressions
- `CONSTRAIN UNIQUE` has greater support in `ON CONFLICT` logic (whereas `UNIQUE INDEX` only supports `INSERT ... ON CONFLICT` I think).

Reply:
Yeah, I was looking at this today. I think that the `ON CONFLICT` capabilities are actually the same, from what I was reading, at least for recent releases? Given that the separate indices can be dropped and added independently, I think I have a slight preference for them, but I don't think it makes much difference one way or the other.